Test reading `__cuda_array_interface__` inferred strides.

gchanan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#31451 Test reading `__cuda_array_interface__` inferred strides.**

The PR that fixed this, https://github.com/pytorch/pytorch/pull/24947, didn't add a test.

Fixes: https://github.com/pytorch/pytorch/issues/31443

Differential Revision: [D19170020](https://our.internmc.facebook.com/intern/diff/D19170020)

gchanan committed:
Test reading `__cuda_array_interface__` inferred strides.

The PR that fixed this, https://github.com/pytorch/pytorch/pull/24947, didn't add a test.

Fixes: https://github.com/pytorch/pytorch/issues/31443

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit e398b16:
* **1/1** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=e398b164d903db36efa24fe8f6478838816c1dd6).

### :detective: 1 new failure recognized by patterns
The following build failures don't appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4010883) pytorch_linux_xenial_py3_6_gcc5_4_test (1/1)
**Step:** "Test" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=77118742) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=77118742))
<details>
<summary>
<code>Dec 19 01:02:43 RuntimeError: test_dist_autograd_spawn failed!</code>
</summary>

```
Dec 19 01:02:42 Ran 49 tests in 61.334s 
Dec 19 01:02:42  
Dec 19 01:02:42 FAILED (failures=1, skipped=2) 
Dec 19 01:02:42  
Dec 19 01:02:42 Generating XML reports... 
Dec 19 01:02:43 Traceback (most recent call last): 
Dec 19 01:02:43   File "test/run_test.py", line 456, in <module> 
Dec 19 01:02:43     main() 
Dec 19 01:02:43   File "test/run_test.py", line 449, in main 
Dec 19 01:02:43     raise RuntimeError(message) 
Dec 19 01:02:43 RuntimeError: test_dist_autograd_spawn failed! 
Dec 19 01:02:43 + cleanup 
Dec 19 01:02:43 + retcode=1 
Dec 19 01:02:43 + set +x 
Dec 19 01:02:43 =================== sccache compilation log =================== 
Dec 19 01:02:43 ERROR:sccache::server: Compilation failed: Output { status: ExitStatus(ExitStatus(256)), stdout: "", stderr: "/tmp/torch_extensions/test_compilation_error_formatting/main.cpp: In function \'int main()\':\n/tmp/torch_extensions/test_compilation_error_formatting/main.cpp:2:23: error: expected \';\' before \'}\' token\n int main() { return 0 }\n                       ^\n" } 
Dec 19 01:02:43  
Dec 19 01:02:43 =========== If your build fails, please take a look at the log above for possible reasons =========== 
Dec 19 01:02:43 Compile requests                115 
Dec 19 01:02:43 Compile requests executed        25 
Dec 19 01:02:43 Cache hits                       18 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

zou3519 commented on test/test_numba_integration.py:
```diff
@@ -309,6 +309,30 @@ def test_from_cuda_array_interface(self):
+    @unittest.skipIf(not TEST_CUDA, "No cuda")
+    @unittest.skipIf(not TEST_NUMBA_CUDA, "No numba.cuda")
+    def test_from_cuda_array_interface_inferred_strides(self):
+        """torch.as_tensor(numba_ary) should have correct inferred (contiguous) strides"""
```
Nit: I think `numba_ary` read better as `numba_array`; I haven't seen too many usages of "ary" to mean array in the past

gchanan replied:
I agree -- I was just following the style of the rest of the test here (which seems useful, since this test should just be combined with others once we make the test less strict).

zou3519 commented on test/test_numba_integration.py:
```diff
@@ -309,6 +309,30 @@ def test_from_cuda_array_interface(self):
+        """torch.as_tensor(numba_ary) should have correct inferred (contiguous) strides"""
+        # This could, in theory, be combined with test_from_cuda_array_interface but that test
+        # is overly strict: it checks that the exported protocols are exactly the same, which
+        # cannot handle differening exported protocol versions.
```
nit: s/differening/differing

gchanan committed:
Update on "Test reading `__cuda_array_interface__` inferred strides."


The PR that fixed this, https://github.com/pytorch/pytorch/pull/24947, didn't add a test.

Fixes: https://github.com/pytorch/pytorch/issues/31443

Differential Revision: [D19170020](https://our.internmc.facebook.com/intern/diff/D19170020)

[ghstack-poisoned]
